### PR TITLE
Mention that -D warnings will deny ALL warnings, not just clippy warnings

### PR DESCRIPTION
**Pull request by sunjay**
If we pass `-D warnings` to clippy, it causes the build to fail if there are *any* warnings, even the ones not generated by clippy. This isn't immediately obvious when you're looking at this as someone just setting up clippy, so people might not expect this nor know how to fix it. I've added a few sentences in the README to help anyone who runs into this.

These docs are useful for anyone setting up clippy warnings to be denied during CI, but still otherwise want rustc warnings to be allowed.

I could have also changed `-D warnings` to `-D clippy::all` in the Travis configurations themselves, but I wasn't sure what you would prefer to have people use as the default.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -125,6 +125,11 @@ before_script:
    # etc
 ```
 
+Note that adding `-D warnings` will cause your build to fail if **any** warnings are found in your code.
+That includes warnings found by rustc (e.g. `dead_code`, etc.). If you want to avoid this and only cause
+an error for clippy warnings, use `#![deny(clippy::all)]` in your code or `-D clippy::all` on the command
+line. (You can swap `clippy::all` with the specific lint category you are targetting.)
```
Comment (flip1995) [inline README.md]:
```suggestion
line. (You can swap `clippy::all` with the specific lint category you are targeting.)
```

**Comment by flip1995** [comment]
We highly prefer `-D warnings` in travis. In Rust code you don't want to have **any** warnings in your crate, since these will lead to bugs.

E.g.
```rust
enum Foo {
    A,
    B,
}
//...
match some_foo {
    A => (),
    B => (),
}
```
will only produce a warning, telling you that `B => (),` is unreachable, since the first arm is a variable binding. It also suggests that you should use `Foo::A` instead.

`dead_code` warnings won't produce bugs like the one above, but if you have to many `dead_code` warnings in your code, you may miss warnings like the one above. Queue 2h of debugging. Been there, done that 😄 

---
TL;DR: I'm willing to merge this with the typo fixed, but you should prevent any warnings from the rust compiler in production.

**Comment by sunjay** [comment]
Typo is fixed! Yes I agree about the warnings in general, but we should still warn/inform people about what we're advising them to do. 

Thanks for reviewing this! :)
